VideoLocker/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -150,6 +158,31 @@ public void onApplicationLaunchedFromBackground(){
     }
 
 
+    /**
+     * if app is launched from upgrading, we need to resync with parse server.
+     * @param context
+     */
+    private void tryToUpdateParseForAppUpgrade(Context context){
```

Comment:
@aleffert  if we do the check each time app is brought to the front, it maybe too much? i am not sure.  maybe we can add a timestamp in addition to the event when app is brought to the front.